sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -811,14 +811,15 @@ static ssize_t _add_opt_pkt(coap_pkt_t *pkt, uint16_t optnum, const uint8_t *val
     return optlen;
 }
 
-ssize_t coap_opt_add_string(coap_pkt_t *pkt, uint16_t optnum, const char *string,
-                           char separator)
+ssize_t coap_opt_add_chars(coap_pkt_t *pkt, uint16_t optnum, const char *chars,
+                           size_t chars_len, char separator)
 {
-    size_t unread_len = strlen(string);
+    size_t unread_len = chars_len;
```

Comment:
Since we have chars_len now, do we need to define a separate variable to decrement the length? Why not just reuse chars_len? What is your experience with the rest of the RIOT code base for this situation? Maybe add a comment so the reader doesn't miss this usage? I don't see any guidance in the coding guidelines.